gadopt/time_stepper.py

```diff
@@ -117,120 +124,209 @@ def advance(
         self.get_final_solution()
 
 
-class ERKGeneric(RungeKuttaTimeIntegrator):
-    """Generic explicit Runge-Kutta time integrator.
+class IrksomeIntegrator(TimeIntegratorBase):
+    """Time integrator using Irksome as the backend.
 
-    Implements the Butcher form. All terms in the equation are treated explicitly.
+    This class wraps Irksome's TimeStepper while maintaining G-ADOPT's API
+    for compatibility with our existing code.
 
-    Arguments:
-      equation: G-ADOPT equation to solve
-      solution: Firedrake function reperesenting the equation's solution
-      dt: Integration time step
-      solution_old: Firedrake function representing the equation's solution
+    Args:
+        equation: G-ADOPT equation to integrate
+        solution: Firedrake function representing the equation's solution
+        dt: Integration time step (Firedrake Constant or float)
+        butcher: Irksome Butcher tableau (e.g., GaussLegendre, RadauIIA)
+        stage_type: Type of stage formulation (e.g., "deriv")
+        solution_old: Firedrake function representing the equation's solution
                       at the previous timestep
-      solver_parameters: Dictionary of solver parameters provided to PETSc
-      strong_bcs: List of Firedrake Dirichlet boundary conditions
-
+        strong_bcs: List of Firedrake Dirichlet boundary conditions
+        bc_type: Boundary condition type for Irksome ("DAE" or "ODE")
```

Comment:
Do we support `EquationBC` as part of `strong_bcs`? If yes, best to specify that such a condition is only compatible with `bc_type="DAE"`.